### ISSUE TITLE
IsolatedStorageException: Unable to create the store directory

### ISSUE THREAD

**Issue by tkarpinski** [andrewdavey/cassette#6]
Create and Publishn MVC3 app with Cassette.Web 0.5.2 to a Server 2008 R2 box with standard permissions (i.e. nothing's been modified).  The following error occurs without any additional code changes outside of installing the NuGet Package
### Unable to create the store directory. (Exception from HRESULT: 0x80131468)

Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

```
Exception Details: System.IO.IsolatedStorage.IsolatedStorageException: Unable to create the store directory. (Exception from HRESULT: 0x80131468)
```
### Source Error:

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.  

Stack Trace: 

```
[IsolatedStorageException: Unable to create the store directory. (Exception from HRESULT: 0x80131468)]
   Cassette.Web.ExceptionCachedManager.CreatePageAssetManager(HttpContextBase httpContext) +92
   Cassette.Web.CassetteHttpModule.<Init>b__0(Object sender, EventArgs e) +173
   System.Web.SyncEventExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +80
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +266
```

**Comment by andrewdavey** [comment]
Which user account is IIS running under? I need to check if there's a reason it can't access isolated storage.

**Comment by tkarpinski** [comment]
I'm using the standard AppPool - ASP.NET v4.0 - (ApplicationPoolIdentity)

**Comment by andrewdavey** [comment]
Please try pulling the source for Cassette and changing line 21 of Cassette.Web\CassetteApplication.cs to:

`IsolatedStorageFile.GetMachineStoreForAssembly()`

I'm hoping the app pool account can access that location instead.

**Comment by tkarpinski** [comment]
I think that did the trick - the exception is gone.  now I'll move onto using it.

**Comment by andrewdavey** [comment]
Thank you for testing that. I will try to get it into the source this week and update the nuget package.

**Comment by kaatula** [comment]
Hi Andrew,
I ran into the same problem and fixed it in the same way.

Maybe it makes sense to allow this feature to be configurable through config section?

Thanks

**Comment by andrewdavey** [comment]
Cassette v0.5.3 uses `GetMachineStoreForAssembly` so should fix this issue. Can someone please verify. Thanks.

**Comment by kaatula** [comment]
Just after updating to 0.5.3 and deploying (Windows 2008 R2 virtual maching, AppPoolIdentity user, .NET 4) i got some exception, but I believe it's related with my changes in Cassette. I had manually cleared isolated storage in my Users directory, and web site started working just fine.

Just for your info, that's exception which I got (but I still believe it's my problem, not your one):
'''Server Error in '/' Application.
Unable to delete file. 
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: System.IO.IsolatedStorage.IsolatedStorageException: Unable to delete file.

Source Error: 
An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below. 

Stack Trace: 

[IsolatedStorageException: Unable to delete file.]
   Cassette.Web.ExceptionCachedManager.CreatePageAssetManager(HttpContextBase httpContext) in D:\Sandbox\andrewdavey-cassette-dee75e7\src\Cassette.Web\ExceptionCachedManager.cs:28
   Cassette.Web.CassetteHttpModule.<Init>b__0(Object sender, EventArgs e) in D:\Sandbox\andrewdavey-cassette-dee75e7\src\Cassette.Web\CassetteHttpModule.cs:14
   System.Web.SyncEventExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +80
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +266

Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.0.30319.1'''

**Comment by andrewdavey** [comment]
Sorry, it seems I forgot to push Cassette.Web 0.5.3. It's now there. Please update and try it again.

**Comment by ChrisMH** [comment]
0.5.4 Still uses GetMachineStoreForDomain and I am getting this error in IISExpress.

Changing to GetMachineStoreForAssembly fixes the issue.

Am I missing something, or did you decide not to push this change?

**Comment by andrewdavey** [comment]
Erm, that's odd. Perhaps I messed up the nuget build...

I'm very close to releasing the big-rewrite of Cassette (`big-rewrite` branch for those who want a peak). It should have bugs like this squashed and lots of new features to play with. So I hope you don't mind waiting just a little longer :)

**Comment by andrewdavey** [comment]
Fixed in 0.6
https://github.com/andrewdavey/cassette/blob/master/src/Cassette.Web/StartUp.cs#L50
